internal/services/azurestackhci/stack_hci_deployment_setting_resource.go

```diff
@@ -0,0 +1,1365 @@
+					},
+
+					"host_network": {
+						Type:     pluginsdk.TypeList,
+						Required: true,
+						ForceNew: true,
+						MaxItems: 1,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"intent": {
+									Type:     pluginsdk.TypeList,
+									Required: true,
+									ForceNew: true,
+									MinItems: 1,
+									Elem: &pluginsdk.Resource{
+										Schema: map[string]*pluginsdk.Schema{
+											"name": {
+												Type:         pluginsdk.TypeString,
+												Required:     true,
+												ForceNew:     true,
+												ValidateFunc: validation.StringIsNotEmpty,
+											},
+
+											"adapter": {
+												Type:     pluginsdk.TypeList,
+												Required: true,
+												ForceNew: true,
+												MinItems: 1,
+												Elem: &pluginsdk.Schema{
+													Type:         pluginsdk.TypeString,
+													ValidateFunc: validation.StringIsNotEmpty,
+												},
+											},
+
+											"traffic_type": {
+												Type:     pluginsdk.TypeList,
+												Required: true,
+												ForceNew: true,
+												MinItems: 1,
+												Elem: &pluginsdk.Schema{
+													Type: pluginsdk.TypeString,
+													ValidateFunc: validation.StringInSlice([]string{
+														"Compute",
+														"Storage",
+														"Management",
+													}, false),
+												},
+											},
+
+											"adapter_property_override": {
+												Type:     pluginsdk.TypeList,
+												Optional: true,
+												ForceNew: true,
+												MaxItems: 1,
+												Elem: &pluginsdk.Resource{
+													Schema: map[string]*pluginsdk.Schema{
+														"jumbo_packet": {
+															Type:         pluginsdk.TypeString,
+															Optional:     true,
+															ForceNew:     true,
+															ValidateFunc: validation.StringIsNotEmpty,
+														},
+
+														"network_direct": {
+															Type:         pluginsdk.TypeString,
+															Optional:     true,
+															ForceNew:     true,
+															ValidateFunc: validation.StringIsNotEmpty,
+														},
+
+														"network_direct_technology": {
+															Type:         pluginsdk.TypeString,
+															Optional:     true,
+															ForceNew:     true,
+															ValidateFunc: validation.StringIsNotEmpty,
+														},
+													},
+												},
+											},
+
+											"adapter_property_override_enabled": {
+												Type:     pluginsdk.TypeBool,
+												Optional: true,
+												ForceNew: true,
+												Default:  false,
+											},
+
+											"qos_policy_override": {
+												Type:     pluginsdk.TypeList,
+												Optional: true,
+												ForceNew: true,
+												MaxItems: 1,
+												Elem: &pluginsdk.Resource{
+													Schema: map[string]*pluginsdk.Schema{
+														"bandwidth_percentage_smb": {
```

Comment:
this is a percentage so why is it a string?

Reply:
It's plain string format and related to specific OEM hardware setting. https://github.com/Azure/azure-rest-api-specs/blob/c41b76298becce0a8cfac4e682773d5d0c78dcb2/specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/stable/2024-01-01/deploymentSettings.json#L784-L786